services/src/main/java/org/keycloak/services/x509/AbstractClientCertificateFromHttpHeadersLookupFactory.java

```diff
@@ -55,6 +57,8 @@ public void init(Config.Scope config) {
             } else {
                 logger.tracev("{0} was not configured", HTTP_HEADER_CERT_CHAIN_PREFIX);
             }
+            trustProxyVerification = config.getBoolean(TRUST_PROXY_VERIFICATION, false);
```

Comment:
Does it make more sense to have this option only available to nginx factory?

Reply:
I guess it is true by default in the others, but some one might choose to have a trusted and non trusted apache and use the same variable.

Reply:
If we want to do that, I think I would need to override the init, to call super.init then store the variable. I can do that if you want.

Reply:
Yeah, I see your point. But for now, I would keep it only available to nginx and if needed, refactor it later.

Sorry for asking this, you have been waiting for a long time already :) But just this one and we are good to go.

Reply:
no worries, on it

Reply:
@pedroigor here you go